test/benchmark/main.cc

```diff
@@ -35,6 +36,12 @@ int main(int argc, char** argv) {
     return 0;
   }
 
+  // Reduce logs so benchmark output is readable.
+  static auto* lock = new Thread::MutexBasicLockable();
```

Comment:
Rather than newing these and putting them in statics, just declare them on the stack.

Reply:
I have noticed multiple benchmarks doing this pattern and some that are not absolutely spewing logs (like the fancy log benchmark itself). Instead of doing this in every benchmark, can we setup logging in the common benchmark main and default to warn? And then individual benchmarks can override if necessary? cc @antoniovicente who has been working on the benchmark main stuff most recently. Thank you!

Reply:
This is the shared main that should be used by all benchmarks.  Capping logging to warn here makes sense to me.

+1 to jmarantz suggestion to change to:
Thread::MutexBasicLockable lock;
Logger::Context logging_context{spdlog::level::warn, Logger::Context::getFancyLogFormat(), lock, false};

Reply:
Sorry I wasn't paying attention. I'm pretty sure I saw some other benchmarks setting up logging outside of main. Can you potentially check to see if this is duplicated elsewhere?

Reply:
There are some benchamarks that adjust logging.  I'll make a note to do a pass to get that cleaned up.

Reply:
Sure sounds good. Thank you.

Reply:
Modified to be stack allocated. This one is in benchmark main as pointed out above. I assume @antoniovicente will go through the other benchmarks and clean them up, not sure if I should do that in this PR.